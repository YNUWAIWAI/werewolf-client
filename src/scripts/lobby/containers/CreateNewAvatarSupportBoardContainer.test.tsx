import * as React from 'react'
import {CreateNewAvatarSupportBoard} from '../components/organisms/CreateNewAvatarSupportBoard'
import {CreateNewAvatarSupportBoardContainer} from './CreateNewAvatarSupportBoardContainer'
import {IntlProviderContainer} from './IntlProviderContainer'
import {Provider} from 'react-redux'
import {createNewRobotAvatar} from '../actions'
import fakeStore from './fakeStore'
import {mount} from 'enzyme'

const data = {
  '4': {
    A: false,
    B: false,
    C: false
  },
  '5': {
    A: false,
    B: false,
    C: false
  },
  '6': {
    A: false,
    B: false,
    C: false
  },
  '7': {
    A: false,
    B: false,
    C: false
  },
  '8': {
    A: false,
    B: false,
    C: false
  },
  '9': {
    A: false,
    B: false,
    C: false
  },
  '10': {
    A: false,
    B: false,
    C: false
  },
  '11': {
    A: false,
    B: false,
    C: false
  },
  '12': {
    A: false,
    B: false,
    C: false
  },
  '13': {
    A: false,
    B: false,
    C: false
  },
  '14': {
    A: false,
    B: false,
    C: false
  },
  '15': {
    A: false,
    B: false,
    C: false
  }
}

test('render', () => {
  const store = fakeStore({
    createNewRobotAvatarSupport: {
      data
    }
  })
  const wrapper = mount(
    <Provider store={store} >
      <IntlProviderContainer>
        <CreateNewAvatarSupportBoardContainer />
      </IntlProviderContainer>
    </Provider>
  )

  expect(wrapper.html()).toMatchSnapshot()
})
test('handleSupportBoardClick', () => {
  const store = fakeStore({
    createNewRobotAvatarSupport: {
      data
    }
  })
  const dispatch = jest.fn()

  store.dispatch = dispatch
  const wrapper = mount(
    <Provider store={store} >
      <IntlProviderContainer>
        <CreateNewAvatarSupportBoardContainer />
      </IntlProviderContainer>
    </Provider>
  )

  wrapper.find(CreateNewAvatarSupportBoard).props().handleSupportBoardClick()
  expect(dispatch).toHaveBeenCalledTimes(1)
  expect(dispatch).toHaveBeenCalledWith(createNewRobotAvatar.showSupportSelect())
})
