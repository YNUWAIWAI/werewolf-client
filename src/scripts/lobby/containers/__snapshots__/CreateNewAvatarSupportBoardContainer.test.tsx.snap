// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `"<div class=\\"lo--create-new-avatar--support-board\\"><div class=\\"lo--create-new-avatar--support-board--cell\\"># of players</div><div class=\\"lo--create-new-avatar--support-board--cell\\">A</div><div class=\\"lo--create-new-avatar--support-board--cell\\">B</div><div class=\\"lo--create-new-avatar--support-board--cell\\">C</div><div class=\\"lo--create-new-avatar--support-board--cell\\">15</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">14</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">13</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">12</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">11</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">10</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">9</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">8</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">7</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">6</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">5</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\">4</div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div><div class=\\"lo--create-new-avatar--support-board--cell\\"></div></div>"`;
